contrib/upgrade_python-mastodon.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+# upgrade the python-mastodon module if Debian/RPM package was outdated
+# -----------------------------------------------------------------------------
+# this is intended for the case your distribution ships with an outdated version
+# of the python-mastodon package (pre-1.5.1). It will simply replace the module
+# scripts where your packaging placed them.
+#
+# © 2022 Izzy <izzysoft AT qumran DOT org>; GPL-3.0-or-later
+
+# -----------------------------------------------------------------------------
+# define temp dir and exit-on-error
+function cleanup() {
+  echo
+  echo "Something went wrong, aborting."
+  rm -rf $tmpdir
+  exit 1
+}
+
+tmpdir=$(mktemp -d)
+trap cleanup ERR
+
+
+# ==============
+# ===[ Main ]===
+# ==============
+# -----------------------------------------------------------------------------
+# check whether we need to update python3-mastodon
+echo
+echo "Let's see if your distribution had the recent version of python3-mastodon…"
+if [[ -f /etc/debian_versions ]]; then
+  mver="$(dpkg -l python3-mastodon |tail -n 1 |awk '{print $3}')"
+elif [[ -f /etc/redhat_release || -f /etc/fedora-release ]]; then
+  mver="$(yum info python3-Mastodon |grep -Ei '^Version' |awk '{print $3}')"
+else
+  read -n 1 -p "Could not find out whether your system is DEB or RPM based. Continue anyway? (y/n) " REPLY
```

Comment:
Find out sounds like a translation of herausfinden. I think "Could not determine whether …" might sound better.
In terms of usability, I think we should say what happens next before asking the user whether to continue.
"If you continue now, the package ... will be installed in ... which is the default directory for pip ..." or something like that.